pkg/merkletree/tree.go

```diff
@@ -0,0 +1,173 @@
+package merkletree
+
+import (
```

Comment:
Here's the implementation of the Merkle Tree. The to most important design decisions are describe in the Readme ([link](https://github.com/tablelandnetwork/go-tableland/pull/455/files#diff-8b60b48f44e6be8099f3885f4dc07f54dd625fe46d22ece1538fd780c907aecaR5-R8)). 

I think the implementation is very straightforward and the comments in the code should be enough